src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -783,6 +784,30 @@ protected void addErrorForUpdatingItem(Piece piece, Throwable e) {
     }
   }
 
+  protected Map<String, Piece> getProcessedPiecesForPoLine(String poLineId, Map<String, List<Piece>> piecesGroupedByPoLine) {
+    return StreamEx
+      .of(piecesGroupedByPoLine.getOrDefault(poLineId, Collections.emptyList()))
+      .toMap(Piece::getId, piece -> piece);
+  }
+
+  protected Map<String, Integer> getEmptyResultCounts() {
+    Map<String, Integer> resultCounts = new HashMap<>();
+    resultCounts.put(ProcessingStatus.Type.SUCCESS.toString(), 0);
+    resultCounts.put(ProcessingStatus.Type.FAILURE.toString(), 0);
+    return resultCounts;
+  }
+
+  protected StreamEx<Piece> extractAllPieces(Map<String, List<Piece>> piecesGroupedByPoLine) {
+    return StreamEx.ofValues(piecesGroupedByPoLine).flatMap(List::stream);
+  }
+
+  protected Map<String, List<String>> mapTenantIdsToItemIds(Map<String, List<Piece>> piecesGroupedByPoLine, RequestContext requestContext) {
```

Comment:
This is much like to what I did in `getItemRecrods`: lines 544-551. Maybe that place can reuse this?